src/controller/TypedReadCallback.h

```diff
@@ -108,12 +108,20 @@ class TypedReadAttributeCallback final : public app::ReadClient::Callback
 
     void OnDeallocatePaths(chip::app::ReadPrepareParams && aReadPrepareParams) override
     {
-        VerifyOrDie(aReadPrepareParams.mAttributePathParamsListSize == 1 &&
-                    aReadPrepareParams.mpAttributePathParamsList != nullptr);
-        chip::Platform::Delete<app::AttributePathParams>(aReadPrepareParams.mpAttributePathParamsList);
+        if (aReadPrepareParams.mAttributePathParamsListSize == 1 && aReadPrepareParams.mpAttributePathParamsList != nullptr)
+        {
+            chip::Platform::Delete<app::AttributePathParams>(aReadPrepareParams.mpAttributePathParamsList);
+        }
 
-        VerifyOrDie(aReadPrepareParams.mDataVersionFilterListSize == 1 && aReadPrepareParams.mpDataVersionFilterList != nullptr);
-        chip::Platform::Delete<app::DataVersionFilter>(aReadPrepareParams.mpDataVersionFilterList);
+        if (aReadPrepareParams.mEventPathParamsListSize == 1 && aReadPrepareParams.mpEventPathParamsList != nullptr)
```

Comment:
There will never be event paths here, right?  This callback is only used for attribute reads and subscriptions.  Event reads/subscriptions use TypedReadEventCallback.

Please remove the event bits from here.

Reply:
Thanks for notice. Then VerifyOrDie should be used for mpAttributePathParamsList above.